### PR TITLE
NTNet improvements!

### PR DESCRIPTION
**Pull request by silicons**
I was told to split #36386 
:cl:
rscadd: Circuit ntnet components buffed. Added a new low level ntnet component that can send custom data instead of just the two plaintext and one passkey format, which things will use by default. Ntnet now uses a list for their data instead of three variables. they also have lowered complexity for the now weakened normal network component, and has lower cooldowns.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Kind of hate how this will probably break existing circuits

**Comment by kevinz000** [comment]
@PKPenguin321 it's fully backwards compatible atleast to my knowledge unless i missed something

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -61,6 +61,10 @@
 	set_id(suffix || id || "#[mulebot_count]")
 	suffix = null
 
+/mob/living/simple_animal/bot/mulebot/ComponentInitialize()
+	. = ..()
```
[the remark this answers is not in the document]
Comment (kevinz000) [reply code/modules/mob/living/simple_animal/bot/mulebot.dm]: i left that in earlier

File: code/modules/integrated_electronics/subtypes/input.dm [hunk 2]
```diff
@@ -706,17 +707,60 @@
...
 
 /obj/item/integrated_circuit/input/ntnet_recieve(datum/netdata/data)
 	set_pin_data(IC_OUTPUT, 1, data.sender_id)
-	set_pin_data(IC_OUTPUT, 2, data.plaintext_data)
-	set_pin_data(IC_OUTPUT, 3, data.plaintext_data_secondary)
-	set_pin_data(IC_OUTPUT, 4, data.encrypted_passkey)
+	set_pin_data(IC_OUTPUT, 2, data.data["data"])
+	set_pin_data(IC_OUTPUT, 3, data.data["data_secondary"])
+	set_pin_data(IC_OUTPUT, 4, data.data["encrypted_passkey"])
+	set_pin_data(IC_OUTPUT, 5, data.broadcast)
```
[the remark this answers is not in the document]
Comment (kevinz000) [reply code/modules/integrated_electronics/subtypes/input.dm]: sent to every ntnet device.

**Comment by Electronics** [comment]
Can we please have some basic NTNet info put on the wiki? As it is I have no idea how this works or how it's useful in the game I don't really feel like trawling through the code to find out.

**Comment by PKPenguin321** [comment]
You really don't need to code dive for it. As it stands it's almost exclusively used by integrated circuits. Objects that use NTNet have an address. NTNet senders can take a target address, a text field, a secondary text field, and a passkey field, all of which are just strings. Literally all it does is send a string from one device to another device.

**Pull request merged**